src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Map/AuthorMap/AuthorMap.xaml.cs

```diff
@@ -0,0 +1,497 @@
+                    SaveProgressBar.Visibility = Visibility.Visible;
+
+                    // This is not the initial save, call SaveAsync to save changes to the existing portal item
+                    await _myMap.SaveAsync();
+                    MessageBox.Show("Saved changes to '" + _myMap.PortalItem.Title + "'", "Updates Saved");
+                }
+                catch(Exception ex)
+                {
+                    MessageBox.Show("Unable to save map updates: " + ex.Message);
+                }
+                finally
+                {
+                    // Hide the progress bar
+                    SaveProgressBar.Visibility = Visibility.Hidden;
+                }
+            }
+        }
+
+        private void ClearMap(object sender, RoutedEventArgs e)
+        {
+            // Set the map to null
+            _myMap = null;
+
+            // Show a plain gray map in the map view
+            MyMapView.Map = new Map(Basemap.CreateLightGrayCanvas());
+        }
+        
+        private void UpdateViewExtentLabels()
+        {
+            // Get the current view point for the map view
+            Viewpoint currentViewpoint = MyMapView.GetCurrentViewpoint(ViewpointType.BoundingGeometry);
+            if (currentViewpoint == null) { return; }
+
+            // Get the current map extent (envelope) from the view point
+            Envelope currentExtent = currentViewpoint.TargetGeometry as Envelope;
+            // Project the current extent to geographic coordinates (longitude / latitude)
+            Envelope currentGeoExtent = GeometryEngine.Project(currentExtent, SpatialReferences.Wgs84) as Envelope;
+
+            // Fill the app text boxes with min / max longitude (x) and latitude (y) to four decimal places
+            XMinTextBox.Text = currentGeoExtent.XMin.ToString("0.####");
+            YMinTextBox.Text = currentGeoExtent.YMin.ToString("0.####");
+            XMaxTextBox.Text = currentGeoExtent.XMax.ToString("0.####");
+            YMaxTextBox.Text = currentGeoExtent.YMax.ToString("0.####");
+        }
+
+        #region OAuth helpers
+        private void UpdateAuthenticationManager()
+        {
+            // Register the server information with the AuthenticationManager
+            Esri.ArcGISRuntime.Security.ServerInfo portalServerInfo = new ServerInfo
+            {
+                ServerUri = new Uri(ServerUrl),
+                OAuthClientInfo = new OAuthClientInfo
+                {
+                    ClientId = AppClientId,
+                    RedirectUri = new Uri(OAuthRedirectUrl)
+                },
+                // Specify OAuthAuthorizationCode if you need a refresh token (and have specified a valid client secret)
+                // Otherwise, use OAuthImplicit
+                TokenAuthenticationType = TokenAuthenticationType.OAuthImplicit
+            };
+            AuthenticationManager.Current.RegisterServer(portalServerInfo);
```

Comment:
I might suggest changing the coding pattern for this block of code:
AuthenticationManager.Current.RegisterServer(portalServerInfo);

// Use the OAuthAuthorize class in this project to create a new web view that contains the OAuth challenge handler.
AuthenticationManager.Current.OAuthAuthorizeHandler = new OAuthAuthorize();

// Create a new ChallengeHandler that uses a method in this class to challenge for credentials
AuthenticationManager.Current.ChallengeHandler = new Esri.ArcGISRuntime.Security.ChallengeHandler(CreateCredentialAsync);

To be as follows:
// Get the AuthenticationManager object using the Static AuthenticationManager.Current Method
AuthenticationManager myAuthenticationManager = AuthenticationManager.Current;

// Register the secure server and token endpoints
myAuthenticationManager.RegisterServer(portalServerInfo);

// Use the OAuthAuthorize class in this project to create a new web view that contains the OAuth challenge handler.
myAuthenticationManager.OAuthAuthorizeHandler = new OAuthAuthorize();

// Create a new ChallengeHandler that uses a method in this class to challenge for credentials
myAuthenticationManager.ChallengeHandler = new ChallengeHandler(CreateCredentialAsync);

My rationale is that 'AuthenticationManager.Current' is a Static Method and it creates an AuthenticationManager object. This code will read the same as the rest of the code in the sample if the AuthenticationManager is set to an object.